crates/csharp/src/interface.rs

```diff
@@ -787,9 +1028,9 @@ impl InterfaceGenerator<'_> {
                             .map(|ty| self.type_name_with_qualifier(ty, qualifier))
                             .unwrap_or_else(|| "".to_owned());
                         if name.is_empty() {
-                            return "Task".to_owned();
+                            return "FutureReader".to_owned();
                         } else {
-                            return format!("Task<{name}>");
+                            return format!("FutureReader<{name}>");
```

Comment:
This suggests that `FutureReader` is generic, but it isn't, is it?

Reply:
Correct, removed the generic.